### PR TITLE
sanitized tax amount so that it does not cause "non well formed numer…

### LINKED ISSUE #7434: A non well formed numeric value encountered
`Notice: A non well formed numeric value encountered in C:\wamp64\www\wordpress2\wp-content\plugins\easy-digital-downloads\includes\payments\class-edd-payment.php on line 1417`

When i trying save order detail, it shows this error.

PHP Version: 7.3

EDD Version (or branch): 2.9.19
EDD Software Licensing: 3.6.8

WordPress Version: 5.2.4

Any other relevant information: Running on localhost. And error shows only whrn WP_DEBUG is true.

### PR DESCRIPTION
**Pull request by LisaCee**
sanitized tax amount so that it does not cause "non well formed numeric value" with tax amount over $999, issue#7434

Fixes #7434

Proposed Changes:
1. Ran tax through `edd_sanitize_amount()`
2. Put new sanitized tax amount into totals including tax

For testing:
1. Enable taxes.
2. Complete a payment on the frontend checkout.
3. View the Payment Record in the WordPress dashboard.
4. Modify a cart item's tax amount to be greater than 999, and include a comma in the amount, such as `1,000`
- Verify there are no errors in the error log.

(Thanks, @spencerfinnell )

### CODE REVIEW

File: includes/payments/class-edd-payment.php [hunk 1]
```diff
@@ -1414,6 +1414,7 @@ public function modify_cart_item( $cart_index = false, $args = array() ) {
 		// Format the item_price correctly now
 		$merged_item['item_price']  = edd_sanitize_amount( $merged_item['item_price'] );
 		$new_subtotal               = floatval( $merged_item['item_price'] ) * $merged_item['quantity'];
+		$merged_item['tax']  = edd_sanitize_amount( $merged_item['tax'] );
 		$merged_item['price']       = edd_prices_include_tax() ? $new_subtotal : $new_subtotal + $merged_item['tax'];
```
Comment (spencerfinnell) [inline includes/payments/class-edd-payment.php]:
```suggestion
		$merged_item['item_price'] = edd_sanitize_amount( $merged_item['item_price'] );
		$new_subtotal              = floatval( $merged_item['item_price'] ) * $merged_item['quantity'];
		$merged_item['tax']        = edd_sanitize_amount( $merged_item['tax'] );
		$merged_item['price']      = edd_prices_include_tax() ? $new_subtotal : $new_subtotal + $merged_item['tax'];
```

Some spacing tweaks from https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#indentation

**Comment by spencerfinnell** [comment]
Confirmed original report and verified fix in `issue/7434` -- just a few minor spacing issues but this is good to go otherwise!

File: includes/payments/class-edd-payment.php (hunk 1 above)
Comment (LisaCee) [reply includes/payments/class-edd-payment.php:1418]: @spencerfinnell, did you make this change or should I commit it?

**Comment by cklosowski** [comment]
Looks good to me, once CI passes I'll approve and merge.

**Pull request merged**